### ISSUE TITLE
FID computation involves a large memory footprint

### ISSUE THREAD

**Issue by MartaTintore** [PyTorchLightning/metrics#468]
## 🐛 Bug

When computing the FID score with your implementation I find the following error message:

```
UserWarning: Metric `FID` will save all extracted features in buffer. For large datasets this may lead to large memory footprint. warnings.warn(*args, **kwargs) 
```

As a result, my validation step takes much longer than the training step...

### To Reproduce

I initialize the FID as:

```
from torchmetrics import FID
self.fid = FID(feature=64)
```

And then at every validation step I do:

```
self.fid.update(x, real=True)   
self.fid.update(x_sample, real=False) 
```

x containing a batch of real images and x_sample containing a batch of sampled images

At the end of the validation epoch I do:

`fid = self.fid.compute()`

### Environment

CUDA:
- GPU:
- Quadro GP100
- Quadro GP100
- available: True
- version: 10.2

Packages:
- numpy: 1.19.2
- pyTorch_debug: False
- pyTorch_version: 1.9.0
- pytorch-lightning: 1.4.2
- tqdm: 4.61.0

System:
- OS: Linux
- architecture:
- 64bit
- ELF
- processor: x86_64
- python: 3.8.5
- version: Training accuracy #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020

**Comment by SkafteNicki** [comment]
Hi @MartaTintore,
There is nothing wrong here. We always give people this warning (its not an error) when they initialize the FID class:
https://github.com/PyTorchLightning/metrics/blob/689b2189c6f2aff3968d94d8e5fcfdb85dc5b98a/torchmetrics/image/fid.py#L224-L228
We store the extracted inception features in memory, so if you are evaluating a large number of samples using the FID class it can lead to OOM errors.

That your validation step takes extra time is also expected, as I said all samples are run through an extra inception net to extract the features and the final computation of the score requires the computation of the wasserstein distance which is a fairly expensive operation.

**Comment by MartaTintore** [comment]
The job gets killed and I am not able to compute the metric, anything I could do about it?
